File: frontend/src/types/confirmation.ts
import type { ToolExecutionResult } from './agent';

export interface PendingAction {
  id: number;
  candidate_id?: number | null;
  action_type: string;
  risk_level: string;
  payload?: Record<string, unknown> | null;
  reason?: string | null;
  status: string;
  requested_by?: string | null;
  approved_by?: string | null;
  created_at: string;
  updated_at: string;
}

export interface ConfirmationActionResponse {
  success: boolean;
  message: string;
  pending_action?: PendingAction | null;
  executed_result?: ToolExecutionResult | null;
}

File: backend/app/schemas/confirmation.py
from datetime import datetime
from typing import Any, Dict, Optional

from pydantic import BaseModel, ConfigDict


class PendingActionOut(BaseModel):
    id: int
    candidate_id: Optional[int] = None
    action_type: str
    risk_level: str
    payload: Optional[Dict[str, Any]] = None
    reason: Optional[str] = None
    status: str
    requested_by: Optional[str] = None
    approved_by: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ApproveActionRequest(BaseModel):
    approved_by: str = "HR负责人"


class RejectActionRequest(BaseModel):
    approved_by: str = "HR负责人"
    reason: Optional[str] = None


class ConfirmationActionResponse(BaseModel):
    success: bool
    message: str
    pending_action: Optional[PendingActionOut] = None
    executed_result: Optional[Dict[str, Any]] = None
